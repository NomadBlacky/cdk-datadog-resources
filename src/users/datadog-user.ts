import { CfnResource, Construct } from '@aws-cdk/core';
// eslint-disable-next-line @typescript-eslint/no-require-imports
import camelcaseKeys = require('camelcase-keys');
import { DatadogCredentials } from '../common/properties';

export interface DatadogIAMUserProps {
  /** Credentials for the Datadog API */
  readonly datadogCredentials: DatadogCredentials;

  /** The role of the user */
  readonly accessRole?: string;

  /** Whether or not the user is disabled */
  readonly disabled?: boolean;

  /** User email */
  readonly email?: string;

  /** User handle (a valid email) */
  readonly handle?: string;

  /** User name */
  readonly name?: string;

  /** Whether or not the user is verified */
  readonly verified?: boolean;
}

/**
 * Datadog Application User 1.2.0
 */
export class DatadogIAMUser {
  constructor(scope: Construct, id: string, props: DatadogIAMUserProps) {
    const cfnProperties = camelcaseKeys(props, {
      deep: true,
      pascalCase: true,
    });
    new CfnResource(scope, id, {
      type: 'Datadog::IAM::User',
      properties: { ...cfnProperties },
    });
  }
}
