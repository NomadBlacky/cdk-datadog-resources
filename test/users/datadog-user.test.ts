import { SynthUtils } from '@aws-cdk/assert';
import { Stack } from '@aws-cdk/core';
import { DatadogIAMUser } from '../../src/users/datadog-user';

test('Snapshot test', () => {
  const stack = new Stack();

  new DatadogIAMUser(stack, 'TestUser', {
  	datadogCredentials: {
      apiKey: 'DATADOG_API_KEY',
      applicationKey: 'DATADOG_APP_KEY',
    },
    email: "jane.doe@example.com",
    name: "name_example",
    handle: "title_example",
    disabled: false,
  });
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});