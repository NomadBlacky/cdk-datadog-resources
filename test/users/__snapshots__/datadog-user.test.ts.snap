// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "Resources": Object {
    "TestUser": Object {
      "Properties": Object {
        "DatadogCredentials": Object {
          "ApiKey": "DATADOG_API_KEY",
          "ApplicationKey": "DATADOG_APP_KEY",
        },
        "Disabled": false,
        "Email": "jane.doe@example.com",
        "Handle": "title_example",
        "Name": "name_example",
      },
      "Type": "Datadog::IAM::User",
    },
  },
}
`;
